WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldEditorView.swift

```diff
@@ -0,0 +1,208 @@
+import SwiftUI
+
+struct AztecEditorView: UIViewControllerRepresentable {
+    @Binding var html: String
+
+    func makeUIViewController(context: Context) -> AztecEditorViewController {
+        let viewProperties = EditorViewProperties(navigationTitle: "Navigation title", /* todo replace string */
+                                                  placeholderText: "placeholder text", /* todo replace string */
+                                                  showSaveChangesActionSheet: true)
+
+        let controller = AztecEditorViewController(
+            content: html,
+            product: nil,
+            viewProperties: viewProperties,
+            isAIGenerationEnabled: false
+        )
+
+        return controller
+    }
+
+    func updateUIViewController(_ uiViewController: AztecEditorViewController, context: Context) {
+        // Update the view controller if needed
+    }
+}
+
+struct CustomFieldEditorView: View {
+    @State private var key: String
+    @State private var value: String
+    @State private var isModified: Bool = false
+    @State private var showRichTextEditor: Bool = false
+    @State private var showingActionSheet: Bool = false
+
+    private let initialKey: String
+    private let initialValue: String
+    private let isReadOnlyValue: Bool
+
+
+    /// Initializer for custom field editor
+    /// - Parameters:
+    ///  - key: The key for the custom field
+    ///  - value: The value for the custom field
+    ///  - isReadOnlyValue: Whether the value is read-only or not. To be used if the value is not string but JSON.
+    init(key: String, value: String, isReadOnlyValue: Bool = false) {
```

Comment:
Why not pass the `CustomFieldViewModel` as input here? Any specific reason?

Reply:
This editor will also be used when adding a new custom field, and my thinking was that we don't need all the extra information in `CustomFieldViewModel` for this purpose, and instead just pass empty strings instead. wdyt?

Reply:
I think it's fine to start with, but consider you should not be able to test the editor logic.

Reply:
Good point here, I'll keep in mind for future refactor I think!